.github/workflows/cmake.yml

```diff
@@ -0,0 +1,95 @@
+# from https://github.com/OSGeo/gdal/blob/master/.github/workflows/cmake_builds.yml
+
+name: Build and test with CMake
+
+on:
+    push:
+        paths-ignore:
+            - 'doc/**'
+        branches-ignore:
+            - 'backport**'
+    pull_request:
+        paths-ignore:
+            - 'doc/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+env:
+  CMAKE_UNITY_BUILD: OFF
+
+permissions:
+  contents: read  #  to fetch code (actions/checkout)
+
+jobs:
+
+  build-linux-ubuntu-focal:
+    runs-on: ubuntu-20.04
+    env:
+      CMAKE_OPTIONS: -DPython_LOOKUP_VERSION=3.8 -DUSE_CCACHE=ON
+      cache-name: cmake-ubuntu-focal
+    steps:
+    - name: Checkout GRASS
+      uses: actions/checkout@v3
+    - name: Setup cache
+      uses: actions/cache@v3
+      id: cache
+      with:
+          path: ${{ github.workspace }}/.ccache
+          key: ${{ runner.os }}-${{ env.cache-name }}-${{ github.base_ref }}${{ github.ref_name }}-${{ github.run_id }}
+          restore-keys: |
+            ${{ runner.os }}-${{ env.cache-name }}-${{ github.base_ref }}
+            ${{ runner.os }}-${{ env.cache-name }}
+    - name: Install CMake 3.9.1
+      run: |
+        cd ${GITHUB_WORKSPACE}
+        wget https://github.com/Kitware/CMake/releases/download/v3.9.1/cmake-3.9.1-Linux-x86_64.tar.gz
+        tar xzf cmake-3.9.1-Linux-x86_64.tar.gz
+        echo "CMAKE_DIR=$GITHUB_WORKSPACE/cmake-3.9.1-Linux-x86_64/bin" >> $GITHUB_ENV
+        echo "$GITHUB_WORKSPACE/cmake-3.9.1-Linux-x86_64/bin" >> $GITHUB_PATH
+    - run: |
+        cmake --version
+    - name: Install dependencies
+      run: |
+        sudo add-apt-repository -y ppa:ubuntugis/ubuntugis-unstable
+        sudo apt-get update
+        sudo apt-get install -y -q build-essential \
+         flex make bison gcc libgcc1 g++ ccache \
+         python3 python3-dev \
+         python3-opengl python3-wxgtk4.0 \
+         python3-dateutil libgsl-dev python3-numpy \
+         wx3.0-headers wx-common libwxgtk3.0-gtk3-dev \
+         libwxbase3.0-dev libbz2-dev zlib1g-dev gettext \
+         libtiff5-dev libpnglite-dev libcairo2 libcairo2-dev \
+         sqlite3 libsqlite3-dev \
+         libpq-dev libreadline6-dev libfreetype6-dev \
+         libfftw3-3 libfftw3-dev \
+         libboost-thread-dev libboost-program-options-dev libpdal-dev\
+         subversion libzstd-dev checkinstall \
+         libglu1-mesa-dev libxmu-dev \
+         ghostscript wget -y make ccache g++ 
+        sudo apt-get update -y
+    - name: Configure ccache
+      run: |
+          echo CCACHE_BASEDIR=$PWD >> ${GITHUB_ENV}
+          echo CCACHE_DIR=$PWD/.ccache >> ${GITHUB_ENV}
+          echo CCACHE_MAXSIZE=250M >> ${GITHUB_ENV}
+          ccache -z
+      working-directory: ${{ github.workspace }}
+    - name: Configure
+      run: |
+        mkdir -p $GITHUB_WORKSPACE/build
+        cd $GITHUB_WORKSPACE/build
+        cmake ${CMAKE_OPTIONS} ..
```

Comment:
> If the mlocate and updatedb is not needed

No, they aren't - just a debugging leftover.

Reply:
> If the mlocate and updatedb is not needed, these three lines could be simply the following:
> 
> ```shell
> cmake [<options>] -S <path-to-source> -B <path-to-build>
> ```
> 
> ([New in version 3.13](https://cmake.org/cmake/help/latest/manual/cmake.1.html#generate-a-project-buildsystem) (released november 2018, already 4 years ago, python version released at that date was 3.7.1)

The CI currently fails with this change:

```
Run cmake ${CMAKE_OPTIONS} -S $GITHUB_WORKSPACE -B $GITHUB_WORKSPACE/build
CMake Error: The source directory "/home/runner/work/grass/grass/build" does not exist.
```

Reply:
I see it now, and it makes sense from the comment about the old cmake 3.9.1 version downloaded. I made a branch based of this PR branch and was able to test faster, and if we remove the step that downloads the old version and sets it to path, then the default version of 3.25.1 is used, without problem. Once my initial PR to your branch is merged (to update to upstream/main) I can add a comment to suggest the change.

Reply:
> Once my initial PR to your branch is merged (to update to upstream/main) I can add a comment to suggest the change.

Do you refer to https://github.com/neteler/grass/pull/1?
It contains a lot of unrelated changes ("Files changed 3,090") - not sure how to deal with it.

Reply:
It is merging main into the pr branch. It reduces the diff with main in fact. The big part of the large file count is because of the clang formatting commits that created som conflicts here.